python/GafferOSLTest/ShadingEngineTest.py

```diff
@@ -479,6 +479,26 @@ def testReadV2fUserData( self ) :
 				imath.Color3f( p["P"][i].x, p["P"][i].y, 0 )
 			)
 
+	def testCanReadStringData( self ):
+		IECore.waitForDebugger()
```

Comment:
Let's omit this line. I can see the benefit in having it in `IECore.TestCase.setUp()` so it's optionally available for _any_ invocation of `gaffer test SomeTestCase.OrOther`, but I don't see much benefit in leaving it lying around in individual tests.

While we're talking about this, are you aware of `env GAFFER_DEBUG=1 gaffer ...`? That should [launch Gaffer in a debugger](https://github.com/GafferHQ/gaffer/blob/master/bin/gaffer#L264) automatically. Might not be useful at IE because we bypass the standard wrapper there, but I think there's an `ieEnvDebug` equivalent?

Reply:
Removed *waitForDebugger* call in deb3770.

> I can see the benefit in having it in IECore.TestCase.setUp() so it's optionally available for any invocation of gaffer test SomeTestCase.OrOther, but I don't see much benefit in leaving it lying around in individual tests.

I might remove the python binding and put this as early as I can in the initialisation of the IECore shared library.  

> While we're talking about this, are you aware of env GAFFER_DEBUG=1 gaffer ...? That should launch Gaffer in a debugger automatically. Might not be useful at IE because we bypass the standard wrapper there, but I think there's an ieEnvDebug equivalent?

Sadly, as far as I know, this only runs gdb which is in practice an much less productive than a graphical debugger. I don't know of a way of injecting any GUI based debugger using the environment variables in the various wrappers either in public gaffer or internal to IE. Also, I'm not sure if this will work with the various DCCs, hence this crazy looking loop to wait for the user.

I've also tried the following without success:
* Setting gdb to follow child processes and launching ```scons test``` using a graphical debugger
* gdbserver and remote debbuging.

Reply:
> I might remove the python binding and put this as early as I can in the initialisation of the IECore shared library.

Sounds good!

> Sadly, as far as I know, this only runs gdb which is in practice an much less productive than a graphical debugger. I don't know of a way of injecting any GUI based debugger using the environment variables in the various wrappers either in public gaffer or internal to IE.

Ah, of course - thanks for the explanation. I still need a lesson in proper debugging - I know `bt` and that's about it...